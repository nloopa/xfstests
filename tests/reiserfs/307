#! /bin/bash
# FS QA Test No. 307
#
# Check if ctime is updated and written to disk after setfacl
#
# Regression test for the following extN commits
# c6ac12a ext4: update ctime when changing the file's permission by setfacl
# 30e2bab ext3: update ctime when changing the file's permission by setfacl
# 523825b ext2: update ctime when changing the file's permission by setfacl
#
# Based on test 277
#
#-----------------------------------------------------------------------
# Copyright (c) 2013 Red Hat, Inc.  All Rights Reserved.
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU General Public License as
# published by the Free Software Foundation.
#
# This program is distributed in the hope that it would be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program; if not, write the Free Software Foundation,
# Inc.,  51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
#-----------------------------------------------------------------------
#

seq=`basename $0`
seqres=$RESULT_DIR/$seq
echo "QA output created by $seq"

status=1	# failure is the default!
trap "_cleanup; exit \$status" 0 1 2 3 15

_cleanup()
{
	cd /
	rm -f $testfile
}

# get standard environment, filters and checks
. ./common/rc
. ./common/filter

testfile=$SCRATCH_MNT/testfile.$seq

# real QA test starts here
_supported_fs generic
_supported_os Linux
_require_scratch

echo "Silence is golden"

_scratch_mkfs >/dev/null 2>&1
_scratch_mount >/dev/null 2>&1

touch $testfile
_scratch_remount
ctime1=`stat -c %Z $testfile`

sleep 1
setfacl -m 'u::x,g::x,o::x' $testfile
ctime2=`stat -c %Z $testfile`

_scratch_remount
ctime3=`stat -c %Z $testfile`

if [ "$ctime1" == "$ctime2" ]; then
	echo "error: ctime not updated after setfacl"
elif [ "$ctime1" == "$ctime3" ]; then
	echo "error: on disk ctime not updated"
else
	status=0
fi

exit
