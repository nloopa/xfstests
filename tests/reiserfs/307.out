QA output created by 307
Silence is golden
